### PR TITLE
feat(ctext): Feature/47523 ctext metadata support cognigy

### PR DESCRIPTION
**Pull request by XavierJordaMurria**

### CODE REVIEW

File: src/utils/textSplitter.ts [hunk 1]
```diff
@@ -13,11 +13,10 @@ export async function splitDocs(
 
 	const splitterToUse = options.splitter || defaultSplitter;
 
-	const chunkSize = options.chunkSize || 2000;
+	const chunkSize = options.chunkSize || 1000;
```
Comment (Anurag9589) [inline src/utils/textSplitter.ts:16]: please check default chunk size is 2000

File: src/utils/textSplitter.ts [hunk 2]
```diff
@@ -13,11 +13,10 @@ export async function splitDocs(
...
 
 	// We check if chunkOverlap was defined, because when chunkOverlap is 0, we do not want to default to 200.
-	const chunkOverlap =
-		typeof options.chunkOverlap !== "undefined" ? options.chunkOverlap : 200;
+	const chunkOverlap = 0;
```
Comment (Anurag9589) [inline src/utils/textSplitter.ts:19]: I do not get why we removed this, what is the rationale behind this?
The previous code was correct based on the comment, is there any issue we are facing?

File: src/commands/knowledgeAI/extractCMD.ts [hunk 3]
```diff
@@ -8,5 +8,6 @@ export const extractCMD = async (
 	type: string,
     options: IExtractOptions
 ) => {
+
```
Comment (Anurag9589) [inline src/commands/knowledgeAI/extractCMD.ts:11]: can remove unwanted line here if you want

File: src/lib/knowledgeAI/extract.ts [hunk 4]
```diff
@@ -111,12 +117,21 @@ export const extract = async (type: string, options: IExtractOptions) => {
 
     spinner.stop();
     if (content) {
-        await writeResultsToFile(options.outputFile, content);
+
+        // prepend ctxt version number
+        const cContent = "`version: 1`\n\n" + content;
```
Comment (Anurag9589) [inline src/lib/knowledgeAI/extract.ts:122]: Just a thought, do you think we hardcode the `version: 1` text, I am thinking it is not configurable.
When we introduce a new version, do we always have to change this?

This comment can be completely irrelevant, though. Thanks!

Comment (XavierJordaMurria) [reply src/lib/knowledgeAI/extract.ts:122]: well this, is an npm package, we will need to do a new release anyways,

**Review by Anurag9589 — APPROVED** [review]
Approved!

**Pull request merged**